core/domain/translation_domain.py

```diff
@@ -20,7 +20,248 @@
 
 from core import utils
 
-from typing import Dict
+from typing import Any, Dict, List
+
+
+TRANSLATABLE_CONTENT_FORMAT_HTML = 'html'
+TRANSLATABLE_CONTENT_FORMAT_UNICODE_STRING = 'unicode'
+TRANSLATABLE_CONTENT_FORMAT_SET_OF_NORMALIZED_STRING = (
+    'set_of_normalized_string')
+TRANSLATABLE_CONTENT_FORMAT_SET_OF_UNICODE_STRING = 'set_of_unicode_string'
+
+
+class BaseTranslatableObject:
+    """Base class for all translatable objects."""
+
+    # Private field to track translatable contents in a BaseTranslatableObject.
+    _translatable_contents: Dict[str, TranslatableContent] = {}
+
+    def _register_all_translatable_fields(
+        self
+    ) -> None:
+        """Base class method to register all translatable fields.
+
+        Raises:
+            NotImplementedError. The derived child class must implement the
+                necessary logic to register all translatable fields in an
+                entity.
+        """
+
+        raise NotImplementedError(
+            'Translatable object is not registered, '
+            'by using `_register_all_translatable_fields` method.')
+
+    def _register_translatable_field(
+        self,
+        field_type: str,
+        content_id: str,
+        value: str|List[str]
+    ) -> None:
+        """Base method to register a translatable field in an entity.
+
+        Args:
+            field_type: str. Translatable content type.
+            content_id: str. Content id of translatable content.
+            value: str|list(str). Value of the content which is translatable.
+        """
+        if content_id in self._translatable_contents:
+            # NOTE: Is this expected.
```

Comment:
This comment is for confirmation regarding the Exception message.
@DubeySandeep Can you please confirm whether this message is expected.

(will remove this comment after confirmation.)

Reply:
`A translatable field is already registered with the same content id: %s`, something around this should be better*

Reply:
Updated.